shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -244,6 +244,16 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
   [self updateEditingState];
 }
 
+- (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  if (self.returnKeyType == UIReturnKeyDone && [text isEqualToString:@"\n"]) {
```

Comment:
[-[UITextFieldDelegate textFieldShouldReturn:]](https://developer.apple.com/reference/uikit/uitextfielddelegate/1619603-textfieldshouldreturn) should sort that out I think.